nixos/tests/hostname.nix

```
@@ -39,9 +38,13 @@ let
         # Test if NixOS computes the correct FQDN (either a FQDN or an error/null):
         assert "${getStr nodes.machine.networking.fqdn}" == "${getStr fqdnOrNull}"
 
+        # This is returned by net tools (hostname, domainname) when domain is missing
+        emptyDomainNamePlaceholder="(none)"
+
         # The FQDN, domain name, and hostname detection should work as expected:
-        assert "${fqdn}" == machine.succeed("hostname --fqdn").strip()
-        assert "${optionalString (domain != null) domain}" == machine.succeed("dnsdomainname").strip()
+        assert "${hostName}" == machine.succeed("hostname").strip()
+        assert "${optionalString (domain != null) domain}" == machine.succeed("domainname").replace(emptyDomainNamePlaceholder, "").strip()
```

This took me a few seconds to understand. Better to do less indirections:

```suggestion
        domain = "${if (domain == null) then "(none)" else domain}"
        assert machine.succeed("domainname").strip() == domain
```